tests/system_tests/test_api_endpoints.py

```diff
@@ -0,0 +1,190 @@
+import time
+from typing import Any
+
+import pytest
+import requests
+from fastapi import status
+from pydantic import TypeAdapter
+
+from blueapi.client.rest import BlueapiRestClient, BlueskyRemoteControlError
+from blueapi.service.interface import get_devices, get_plans
+from blueapi.service.model import (
+    DeviceResponse,
+    EnvironmentResponse,
+    PlanResponse,
+    TasksListResponse,
+    WorkerTask,
+)
+from blueapi.worker.event import TaskStatusEnum, WorkerState
+from blueapi.worker.task import Task
+from blueapi.worker.task_worker import TrackableTask
+
+_SIMPLE_TASK = Task(name="sleep", params={"time": 0.0})
+_LONG_TASK = Task(name="sleep", params={"time": 1.0})
+
+rest = BlueapiRestClient()
+
+
+def get_response(
+    url: str, BaseModel: Any, status_code: int = status.HTTP_200_OK
+) -> Any:
+    get_response = requests.get(url)
+    model = TypeAdapter(BaseModel).validate_python(get_response.json())
+    assert get_response.status_code == status_code
+    return model
+
+
+def test_get_plans():
+    assert rest.get_plans() == PlanResponse(plans=get_plans())
+
+
+def test_get_plans_by_name():
+    for plan in get_plans():
+        assert rest.get_plan(plan.name) == plan
```

Comment:
Should: See above for using `get_plans()`, if it were `[]` this test would erroneously pass.

My suggestion here would be to have a parametrised test based on a list of manually curated values: